app/controllers/api/proxy_rules_controller.rb

```diff
@@ -57,19 +46,25 @@ def authorize!
     provider_can_use!(:independent_mapping_rules)
   end
 
-  def find_proxy
-    @proxy = @service.proxy
+  helper_method :destroy_rule_path
+  def destroy_rule_path(service, rule)
```

Comment:
I think this does not belong to the controller.

Reply:
I think that this method is not necessary at all 😄 This method is just an alias for `admin_service_proxy_rule_path` so we could use that one directly instead.

Reply:
@Martouta @guicassolato  the problem is that I wanted to make a different behave from both controllers. if you look in the `_table` partial, I call this method and each controller calls the specific url_path

Reply:
Can we define a helper method (in a normal helper file) and use `controller_name`? Not pretty but better than polluting the controller with view-related stuff.

Reply:
I think you can also do it with https://apidock.com/rails/v4.2.9/ActionDispatch/Routing/PolymorphicRoutes/polymorphic_path

Reply:
```ruby
polymorphic_path [:admin, proxy_rule.owner, ProxyRule]
```

BTW, it should be `admin_proxy_proxy_rule_path ` and not `admin_service_proxy_rule_path`

Reply:
Try

```ruby
owner = proxy_rule.owner
parts = case owner
        when Proxy
          [:admin, owner.service]
        when BackendApi
          [:provider, :admin, owner]
        end
polymorphic_path [*parts, :mapping_rule], id: proxy_rule
```